Realm/ObjectServerTests/RealmServer.swift

```diff
@@ -49,6 +55,282 @@ extension URLSession {
+    static let libDir = buildDir.appendingPathComponent("lib")
+
+    private var path = ["/bin", "/usr/bin"]
+    private var _environment = [String: String]()
+    private var environment: [String: String] {
+        ["PATH": path.joined(separator: ":")].merging(_environment, uniquingKeysWith: { key1, key2 in key1 })
+    }
+
+    @discardableResult
+    func puts(_ commandString: String,
+              currentDirectoryPath: String? = nil) -> (Int32, String) {
+        func createProcesses(_ commandString: String,
+                             process: inout Process,
+                             processes: inout [Process]) {
+            process.currentDirectoryPath = currentDirectoryPath ?? buildDir.absoluteString
+            processes.append(process)
+            var commandAndFlags = commandString.split(separator: " ")
+            process.launchPath = String(commandAndFlags.first!)
+            commandAndFlags.removeFirst()
+            var args = [String]()
+            while commandAndFlags.count > 0 {
+                var argument = commandAndFlags.removeFirst()
+                if argument.starts(with: "\"") {
+                    var nextArg = commandAndFlags.removeFirst()
+                    while nextArg.last != "\"" {
+                        argument += " \(nextArg)"
+                        nextArg = commandAndFlags.removeFirst()
+                    }
+                    argument += " \(nextArg)"
+                }
+                if argument.starts(with: "\'") {
+                    var nextArg = commandAndFlags.removeFirst()
+                    while nextArg.last != "\'" {
+                        argument += " \(nextArg)"
+                        nextArg = commandAndFlags.removeFirst()
+                    }
+                    argument += " \(nextArg)"
+                }
+                if argument != "|" {
+                    args.append(String(argument))
+                } else {
+                    let pipe = Pipe()
+                    let outputPipe = Pipe()
+                    var tailProcess = Process()
+                    createProcesses(commandAndFlags.joined(separator: " "),
+                                    process: &tailProcess,
+                                    processes: &processes)
+                    tailProcess.standardInput = pipe
+                    tailProcess.standardOutput = outputPipe
+                    process.standardOutput = pipe
+                    break
+                }
+            }
+            process.arguments = args
+        }
+        var processes = [Process]()
+        var process = Process()
+        process.environment = environment
+        createProcesses(commandString, process: &process, processes: &processes)
+
+        processes.forEach {
+            $0.launch()
+        }
+        processes.forEach {
+            $0.waitUntilExit()
+        }
+        if let _ = (processes.last!.standardOutput as? Pipe)?.fileHandleForReading.readDataToEndOfFile() {
```

Comment:
Is there a reason this is written twice, and has an unused block?

Reply:
The unused block is some leftover cruft, will remove